lib/src/models/movie_result.dart

```diff
@@ -1,67 +1,67 @@
+import '../../utils/constants.dart';
 import '../resources/movie_strings.dart';
 
 class MovieResult {
-  int _vote_count;
-  int _id;
-  bool _video;
-  var _vote_average;
-  String _title;
-  double _popularity;
-  String _poster_path;
-  String _original_language;
-  String _original_title;
-  List<int> _genre_ids = [];
-  String _backdrop_path;
-  bool _adult;
-  String _overview;
-  String _release_date;
-
-  MovieResult(Map<String, dynamic> result) {
-    _vote_count = result['vote_count'];
-    _id = result['id'];
-    _video = result['video'];
-    _vote_average = result['vote_average'];
-    _title = result['title'];
-    _popularity = result['popularity'];
-    _poster_path = result['poster_path'] != null
-        ? MovieStrings.imageNetwork + result['poster_path']
-        : MovieStrings.imageDefault;
-    _original_language = result['original_language'];
-    _original_title = result['original_title'];
-    for (int i = 0; i < result['genre_ids'].length; i++) {
-      _genre_ids.add(result['genre_ids'][i]);
+  int? voteCount;
+  int? id;
+  bool? video;
+  var voteAverage;
+  String title;
+  var popularity;
+  late String posterPath;
+  String? originalLanguage;
+  String? originalTitle;
+  List<int>? genreIds;
+  String? backdropPath;
+  bool? adult;
+  String overview;
+  String? releaseDate;
+
+  MovieResult({
+    this.voteCount,
+    this.id,
+    this.video,
+    this.voteAverage = Constants.voteAverage,
+    this.title = MovieStrings.emptyString,
+    this.popularity,
+    required this.posterPath,
+    this.originalLanguage,
+    this.originalTitle,
+    this.genreIds,
+    this.backdropPath,
+    this.adult,
+    this.overview = MovieStrings.emptyString,
+    this.releaseDate,
+  });
+
+  factory MovieResult.fromJson(Map<String, dynamic> parsedJson) {
+    var parsedJsonList = parsedJson['genre_ids'] as List;
```

Comment:
Similar to last comment, this can be genresJsonList or something like that

Reply:
done